netsgiro/records.py

```diff
@@ -0,0 +1,245 @@
+import datetime
+import re
+from typing import Optional, Union
+
+import attr
+
+from netsgiro.enums import RecordType, ServiceType
+
+
+def to_date(value: Union[datetime.date, str]) -> Optional[datetime.date]:
+    if isinstance(value, datetime.date):
+        return value
+    if value == '000000':
+        return None
+    return datetime.datetime.strptime(value, '%d%m%y').date()
+
+
+def optional_str(value: str) -> Optional[str]:
+    return value.strip() or None
+
+
+@attr.s
+class Record:
+    service_code = attr.ib()
+    record_type = attr.ib()
+
+    _PATTERN = re.compile(r'^NY.{78}$')
+
+    @classmethod
+    def from_string(cls, line: str) -> 'Record':
+        matches = cls._PATTERN.match(line)
+        if matches is None:
+            raise ValueError('Data did not match data format')
```

Comment:
Add the line and the pattern to the error message. It's not guaranteed that the client has Sentry or some other
traceback-capture present.

`match data format. Pattern: {}, Data: {}`